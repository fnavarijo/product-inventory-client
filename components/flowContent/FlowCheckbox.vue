<template>
  <div>
    <label class="select-none cursor-pointer flex items-center group">
      <input
        :name="name"
        type="checkbox"
        class="checkbox absolute left-0 opacity-0"
        :checked="checked"
        @change="changeValue"
      >
      <div class="checkbox-display h-4 w-4 border border-dark-gray-700 rounded inline-block relative mr-2 group-hover:bg-light-gray-400 transition ease-in-slide duration-200">
        <CheckIcon class="invisible" />
      </div>
      <slot />
    </label>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

import CheckIcon from '@/assets/img/check.svg?inline';

export default Vue.extend({
  name: 'FlowCheckbox',
  components: {
    CheckIcon,
  },
  model: {
    prop: 'checked',
    event: 'change',
  },
  props: {
    name: {
      type: String,
      required: true,
    },
    checked: {
      type: Boolean,
      default: false,
    },
  },
  methods: {
    // TODO: find event type
    changeValue ($event: any): void {
      this.$emit('change', $event.target.checked);
    },
  },
});
</script>

<style scoped>
.checkbox:checked + .checkbox-display {
  @apply bg-dark-gray-700;
}

.checkbox:checked + .checkbox-display > svg {
  @apply visible stroke-white;
}

.checkbox:focus + .checkbox-display {
  @apply shadow-accesible;
}
</style>
