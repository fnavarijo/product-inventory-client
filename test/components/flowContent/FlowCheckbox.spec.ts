import { mount } from '@vue/test-utils';

import FlowCheckbox from '@/components/flowContent/FlowCheckbox.vue';

describe('FlowCheckbox component', () => {
  it('should let select the value of the checkbox', async () => {
    const StubComponent = {
      template: '<div><FlowCheckbox name="check-filter" v-model="checkFlag">CheckFilter</FlowCheckbox></div>',
      components: { FlowCheckbox },
      data () {
        return { checkFlag: true };
      },
    };

    const wrapper = mount(StubComponent);

    const checkbox = wrapper.find('label');
    await checkbox.trigger('click');

    expect(wrapper.vm.$data.checkFlag).toBeFalsy();
  });

  it('should render component with desired design', () => {
    const wrapper = mount(FlowCheckbox, {
      slots: {
        default: 'CheckFilter',
      },
      propsData: {
        name: 'check-filter',
      },
    });

    expect(wrapper.find('label').text()).toContain('CheckFilter');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
