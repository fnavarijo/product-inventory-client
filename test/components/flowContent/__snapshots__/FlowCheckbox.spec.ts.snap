// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowCheckbox component should render component with desired design 1`] = `
<div><label class="select-none cursor-pointer flex items-center group"><input name="check-filter" type="checkbox" class="checkbox absolute left-0 opacity-0">
    <div class="checkbox-display h-4 w-4 border border-dark-gray-700 rounded inline-block relative mr-2 group-hover:bg-light-gray-400 transition ease-in-slide duration-200"><svg class="invisible"></svg></div> CheckFilter
  </label></div>
`;
